### PR TITLE
AWS Image refresh For EBS and EFS

### PR DESCRIPTION
**Pull request by gauravAggrhari**
EFS
=========
**create config**

```
gauravaggrahari@Gauravs-MacBook-Pro aws-efs-csi-driver % ibmcloud sat storage config create --name efs1.2.2p1 --template-name aws-efs-csi-driver --template-version 1.2.2 -p "aws-access-key=<aws-access-key>" -p "aws-secret-access-key=<aws-secret-access-key>" --source-org gauravAggrhari --source-branch master
Creating Satellite storage configuration...
OK
Storage configuration 'efs1.2.2p1' was successfully created with ID 'd7f7bd47-1f5b-4dee-be92-8993c7b73e68'.
```

**create assignment**
```
gauravaggrahari@Gauravs-MacBook-Pro aws-efs-csi-driver % ibmcloud sat storage assignment create --name efsAssignment1.2.2 --group nk-aws --config efs1.2.2p1 
Creating assignment...
OK
Assignment efsAssignment1.2.2 was successfully created with ID 2ae0a417-c849-49ce-b1ff-eb4f11ae5e14.
```

**get pods**

```
gauravaggrahari@Gauravs-MacBook-Pro aws-efs-csi-driver % kubectl get pods -n kube-system | grep efs
efs-csi-controller-7d9d467d56-2q5s9     3/3     Running   0          3m36s
efs-csi-controller-7d9d467d56-9gp57     3/3     Running   0          3m36s
efs-csi-node-5t9lv                      3/3     Running   0          3m36s
efs-csi-node-bpj5z                      3/3     Running   0          3m36s
efs-csi-node-nsmqb                      3/3     Running   0          3m36s
```

**get storageClass**

```
gauravaggrahari@Gauravs-MacBook-Pro aws-efs-csi-driver % kubectl get sc -n kube-system | grep efs
sat-aws-file-gold         efs.csi.aws.com    Delete          Immediate              false                  3m27s
```

```
gauravaggrahari@Gauravs-MacBook-Pro spec % kubectl exec -ti app -- cat  /data/out.txt
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav
gaurav

```

EBS
========

**Get pods in kube-system namepace**
```
gauravaggrahari@Gauravs-MacBook-Pro spec % kubectl get pods -n kube-system | grep ebs
ebs-csi-controller-5669b7cf75-8c4pq     6/6     Running   0          4m14s
ebs-csi-controller-5669b7cf75-wxhpl     6/6     Running   0          4m14s
ebs-csi-node-2njh5                      3/3     Running   0          4m14s
ebs-csi-node-5q5vn                      3/3     Running   0          4m14s
ebs-csi-node-pl2qs                      3/3     Running   0          4m14s
ebs-snapshot-controller-0               1/1     Running   0          4m14s
```

**get storage class**
```
gauravaggrahari@Gauravs-MacBook-Pro spec % kubectl get sc| grep ebs
sat-aws-block-bronze      ebs.csi.aws.com    Delete          WaitForFirstConsumer   true                   15m
sat-aws-block-gold        ebs.csi.aws.com    Delete          WaitForFirstConsumer   true                   15m
sat-aws-block-silver      ebs.csi.aws.com    Delete          WaitForFirstConsumer   true                   15m
```

**get pvc**
```
gauravaggrahari@Gauravs-MacBook-Pro spec % kubectl get pvc                           
NAME                    STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS         AGE
ebs-claim-gaurav-gold   Bound    pvc-9a9c646f-9707-4b0c-a8ce-e4722ca84a8d   20Gi       RWO            sat-aws-block-gold   3m25s
```

**get pod**
```
gauravaggrahari@Gauravs-MacBook-Pro spec % kubectl get pods
NAME   READY   STATUS    RESTARTS   AGE
app    1/1     Running   0          7m12s
```

**Check if volume is mounted : in pod**
```
gauravaggrahari@Gauravs-MacBook-Pro spec % kubectl exec -ti app -- bash
[root@app /]# lsblk
NAME        MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
nvme1n1     259:0    0  100G  0 disk /run/secrets
nvme3n1     259:1    0   50G  0 disk 
nvme2n1     259:2    0  100G  0 disk 
nvme0n1     259:3    0  100G  0 disk 
|-nvme0n1p1 259:4    0    1M  0 part 
`-nvme0n1p2 259:5    0  100G  0 part 
nvme4n1     259:6    0   20G  0 disk /data
```
**in node**
```
sh-4.4# lsblk
NAME        MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
nvme1n1     259:0    0  100G  0 disk /host/var/data
nvme3n1     259:1    0   50G  0 disk 
nvme2n1     259:2    0  100G  0 disk 
nvme0n1     259:3    0  100G  0 disk 
|-nvme0n1p1 259:4    0    1M  0 part 
`-nvme0n1p2 259:5    0  100G  0 part /host
nvme4n1     259:6    0   20G  0 disk /host/var/data/kubelet/pods/662c25d4-590d-4da3-b5a4-6b5cc27d4f1f/volumes/kubernetes.io~csi/pvc-9a9c646f-9707-4b0c-a8ce-e4722ca84a8d/mount
```

### CODE REVIEW

**Comment by gauravAggrhari** [comment]
NA

File: config-templates/aws/aws-ebs-csi-driver/0.9.14/metadata.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      "name": "volume-availability",
+      "value": "zonal"
+    },
+    {
+      "name": "volume-expansion",
+      "value": "true"
+    },
+    {
+      "name": "storageclass",
+      "value": "gold"
```
Comment (mssachan) [inline config-templates/aws/aws-ebs-csi-driver/0.9.14/metadata.json:60]: Update it to "value": "gold,silver,bronze".
Also update for existing version `0.8.2`

File: config-templates/aws/aws-efs-csi-driver/1.2.2/storage-class-parameters.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  }, {
+	"description": "Specify 'true' or 'false' to make the created storage class the default class.",
+	"displayname": "Default",
+	"name": "is-default-class",
+	"default": "false",
+	"required": "false",
+	"category": "config",
+	"type": "text"
+  }, {
+        "description": "Specify the EFS filesystem ID",
```
Comment (mssachan) [inline config-templates/aws/aws-efs-csi-driver/1.2.2/storage-class-parameters.json:17]: Fix indentation issue

File: config-templates/aws/aws-efs-csi-driver/1.2.2/storage-class-parameters.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+[
+  {
+	"description": "The name of the storage class.",
+	"displayname": "Storage Class Name",
+	"name": "name",
+	"required": "true",
+	"category": "config"
+  }, {
+	"description": "Specify 'true' or 'false' to make the created storage class the default class.",
+	"displayname": "Default",
```
Comment (mssachan) [inline config-templates/aws/aws-efs-csi-driver/1.2.2/storage-class-parameters.json:10]: Is default?

File: config-templates/aws/aws-efs-csi-driver/1.2.2/storage-class-parameters.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	"type": "text"
+  }, {
+        "description": "Specify the EFS filesystem ID",
+	"displayname": "EFS Filesystem ID",
+	"name": "fsid",
+	"required": "true",
+	"category": "config"
+
+  },{
+        "description": "Specify direcdtory permissions",
```
Comment (mssachan) [inline config-templates/aws/aws-efs-csi-driver/1.2.2/storage-class-parameters.json:24]: Fix identation issue

File: config-templates/aws/aws-efs-csi-driver/1.2.2/storage-class-parameters.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	"required": "false",
+	"category": "config"
+  }, {
+        "description": "Specify the BasePath",
+	"displayname": "Basepath",
+	"name": "base-path",
+	"required": "false",
+	"category": "config"
+  }
+]
```
Comment (mssachan) [inline config-templates/aws/aws-efs-csi-driver/1.2.2/storage-class-parameters.json:49]: I think all parameters have some default values. 
https://github.com/kubernetes-sigs/aws-efs-csi-driver/tree/master/examples/kubernetes/dynamic_provisioning
Also see if we can improve description msg by giving more detail for each parameter

File: config-templates/aws/aws-efs-csi-driver/1.2.2/metadata.json [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      "name": "provider-url",
+      "value": "https://console.aws.amazon.com/support/home#/"
+    },
+    {
+      "name": "support-msg",
+      "value": "For support contact AWS"
+    },
+    {
+      "name": "provider-logo",
+      "value": "/aws-efs-logo"
```
Comment (nkkashyap) [inline config-templates/aws/aws-efs-csi-driver/1.2.2/metadata.json:24]: Create a `logo` dir and copy `aws-efs-csi-driver.png` under `logo`  refer to https://github.com/IBM/ibm-satellite-storage/tree/develop/config-templates/aws/aws-efs-csi-driver/1.0.0/logo
and update to "value": "logo/aws-efs-csi-driver.png"

File: config-templates/aws/aws-ebs-csi-driver/0.9.14/metadata.json [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      "name": "provider-url",
+      "value": "https://console.aws.amazon.com/support/home#/"
+    },
+    {
+      "name": "support-msg",
+      "value": "For support contact AWS"
+    },
+    {
+      "name": "provider-logo",
+      "value": "/aws-ebs-logo"
```
Comment (nkkashyap) [inline config-templates/aws/aws-ebs-csi-driver/0.9.14/metadata.json:24]: Create a `logo` dir and copy `aws-ebs-csi-driver.png` under `logo`  refer to  https://github.com/IBM/ibm-satellite-storage/tree/develop/config-templates/aws/aws-ebs-csi-driver/0.8.2/logo
and update to `"value": "logo/aws-ebs-csi-driver.png"`

**Review by nkkashyap — CHANGES_REQUESTED** [review]

File: config-templates/aws/aws-efs-csi-driver/1.2.2/metadata.json (hunk 6 above)
Comment (gauravAggrhari) [reply config-templates/aws/aws-efs-csi-driver/1.2.2/metadata.json]: Addressed

File: config-templates/aws/aws-ebs-csi-driver/0.9.14/metadata.json (hunk 7 above)
Comment (gauravAggrhari) [reply config-templates/aws/aws-ebs-csi-driver/0.9.14/metadata.json]: Addressed

**Review by nkkashyap — COMMENTED** [review]
Looks good

**Review by nkkashyap — APPROVED** [review]
Looks good

**Pull request merged**